### PR TITLE
Upgrades app look and feel to Material You

### PR DESCRIPTION
**Pull request by ericdecanini**
This PR shows how the app will look in Material 3, but can't be merged until we have approval from design. Based on that, the requirements and content of this might change
 
## Type of change

- [x] Feature
- [ ] Bugfix
- [ ] Technical
- [ ] Other :

## Content

Upgrades the app's theme to Material 3 (a.k.a Material You)

## Motivation and context

The merge of this PR closes https://github.com/vector-im/element-android/issues/5393

The changelog for 1.5.0 can be found [here](https://github.com/material-components/material-components-android/releases/tag/1.5.0)

Material 1.5.0 introduces the new Material 3 UI design scheme.

The dependency was upgraded without side-effects [in this PR](https://github.com/vector-im/element-android/pull/5392).

This PR includes the full design migration (sans dynamic colors) of the app to Material 3. The components which have been visibly updated are:
- Bottom Navigation Bar
- Chips (e.g. Jump to Unread)
- Popup Menu
- Dialogs
- FAB
- Settings menu (bold preference titles)

Other components have been updated in the code without any user-facing changes due to behavioural changes of UI components in Material 3. These include:
- Text styles
- Settings app bar title alignment

## Screenshots / GIFs

image 
image 
image 
image

## Tests

Smoke test the app and see that designs are either consistent with previous Element app versions or with new Android Material 3 designs.

See that the components above display as expected (whether they are changed or remain the same)

## Tested devices

- [x] Physical
- [ ] Emulator
- OS version(s): Android 10

## Checklist

- [ ] Changes has been tested on an Android device or Android emulator with API 21
- [ ] UI change has been tested on both light and dark themes
- [ ] Accessibility has been taken into account. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#accessibility
- [ ] Pull request is based on the develop branch
- [ ] Pull request includes a new file under ./changelog.d. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#changelog
- [ ] Pull request includes screenshots or videos if containing UI changes
- [ ] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
- [ ] You've made a self review of your PR
- [ ] If you have modified the screen flow, or added new screens to the application, you have updated the test [UiAllScreensSanityTest.allScreensTest()](https://github.com/vector-im/element-android/blob/main/vector/src/androidTest/java/im/vector/app/ui/UiAllScreensSanityTest.kt#L73)

### CODE REVIEW

File: library/ui-styles/src/main/res/values/theme_dark.xml [hunk 1]
```diff
@@ -3,7 +3,7 @@
 
     <!-- DARK THEME COLORS -->
 
-    <style name="Base.Theme.Vector.Dark" parent="Theme.MaterialComponents.NoActionBar">
+    <style name="Base.Theme.Vector.Dark" parent="Theme.Material3.Dark.NoActionBar">
```
Comment (bmarty) [inline library/ui-styles/src/main/res/values/theme_dark.xml:6]: So nice to have explicit `.Dark` in the theme names now!

File: vector/src/main/res/layout/fragment_login_server_url_form.xml [hunk 2]
```diff
@@ -55,7 +55,7 @@
 
             <com.google.android.material.textfield.TextInputLayout
                 android:id="@+id/loginServerUrlFormHomeServerUrlTil"
-                style="@style/Widget.MaterialComponents.TextInputLayout.OutlinedBox.Dense.ExposedDropdownMenu"
+                style="@style/Widget.Material3.TextInputLayout.OutlinedBox.Dense.ExposedDropdownMenu"
```
Comment (bmarty) [inline vector/src/main/res/layout/fragment_login_server_url_form.xml:58]: This was a mistake to directly use `Widget.MaterialComponents.*` in the layout. We should always have override style to our `style_*.xml`. Can I ask you to create the missing files, or add style to existing files following the current naming convention? Thanks!
It apply to all the other changes in layout files.

**Comment by ericdecanini** [comment]
Parking this PR for now whilst awaiting the design team's decision on this.

The PR is left at the state of a full material 3 upgrade with a couple known issues:
- On debug settings, opening a progress dialog will crash the app. This might have implications on other parts of the app
- Some settings dialogs still use the old theme

**Comment by ericdecanini** [comment]
When work is approved and once again ready to work on, either reopen this or use this as reference

**Pull request closed without merging**